### ISSUE TITLE
Preprocess: Generate detector level (within an observation) split flags

### ISSUE THREAD

**Issue by msilvafe** [simonsobs/sotodlib#979]
The [DRP baseline design document](https://docs.google.com/document/d/1cBDbM8KD7DwPp6kfHI_dKUBOC2WwhxltWypEF6deiR0/edit?usp=sharing) outlines a set of null splits written by @chinoney. In order to make atomic maps with these null splits the mapmaker needs to take in a set of flags to choose which detectors (or samples in the case of left/right subscans) to bin into each map. These include:

**Splits based on focalplane/detector (detector-by-detector)**

1. High gain detectors vs. low gain detectors
2. High sensitivity detectors vs. low sensitivity detectors
3. High gain variation detectors vs. low gain variation detectors
4. High time constant detectors vs. low time constant detectors
6. Q-pixel/U-pixel
7. Sense-A/Sense-B for LF
8. “top”' (i.e. Ex^2) detectors vs. “bottom” (i.e. Ey^2) detectors - this is a jargon from PB. this means detectors of crossover or crossunder  
10. High leakage detectors vs. low leakage detectors
11. High 2f (4f) amplitude detectors vs. low amplitude detectors
12. High excess noise detectors at high frequency vs. low noise
14. The Left-half-side pixels vs. right-half-side pixels
15. The Top-half-side pixels vs. bottom-half-side pixels
16. The central pixels vs. the peripheral pixels

**Splits based on subscans**

1. Left vs. right-going sub-scans

We need a new low level Tod-ops module and associated preprocess method to generate these.

**Comment by msilvafe** [comment]
Need to keep this open a bit longer because of the following:
1. We currently have no measure of gain drift. We need to add in a bias step before/after to `det_cal` or estimate this from variations in the HWP A2 but neither of these exist now.
2. I found A/B, Top/Bottom, and pol_angle but not Q/U. I chose to replace Q/U with a split on pol_angle.
4. I'm not sure I know the difference between "high sensitivity detectors vs low sensitivity detectors" and "high excess noise detectors at high frequency vs low noise" so right now I just have one high noise cut based on NEQ.
5. For the top/bottom and left/right focal plane split are those with respect to horizon coordinates or telescope coordinates (i.e. do I compensate for boresight rotation in those or not)? For now I've assumed fixed to telescope boresight coordinate system so that the subset of detectors does not change if boresight rotates.
6. We don't have subscan splitting now so I haven't implemented that but once we merge PR #1028 then this will be easy to add.

However what I've implemented so far in PR #1039 should be more than enough for @DanielBThomas 's Phase 2 and possible 3 plans.
